src/fdc/DiskImageUtils.cc

```diff
@@ -104,6 +104,10 @@ static SetBootSectorResult setBootSector(
 		nbDirEntry = 256;
 		descriptor = 0xF0;
 		nbHiddenSectors = 16;	// override default from above
+		// for a 32MB disk or greater the sectors would be >= 65536
+		// since MSX use 16 bits for this, in case of sectors = 65536
+		// the truncated word will be 0 -> formatted as 320 Kb disk!
+		if (nbSectors > 65535) nbSectors = 65535; // this is the max size for fat12 :-)
```

Comment:
nitpick: `nbSectors = std::min(nbSectors, 65536);`

Though I wonder if moving this line from Diskmanipulator to here doesn't have any other consequences.
For example: before the size of the generated disk image would be truncated to max 65535 sectors. But now we create a possibly larger image with truncated sector count information in the boot sector.

Maybe this is fine, but I wanted to point you to this (maybe unintended) difference.

Maybe instead of silently modifying the sector count, we should give an error then something doesn't fit?

Reply:
That’s intentional, now it is possible to create any size of partition with the diskmanipulator so you can format it with `_FDISK` (absent diskmanipulator support). I thought it was weird that the file was smaller than I indicated.

Modifying the sector count I think is normal, ideally the sector count matches data start + cluster count * cluster size precisely, which is what Nextor’s `_FDISK` actually does. (If not an exact match, it relies on the integer division flooring the cluster count, as well as checking against the max cluster number.)

Reply:
> nitpick: `nbSectors = std::min(nbSectors, 65536);`

Then it becomes `nbSectors = std::min(nbSectors, size_t(65536));` because `std::min` is stubborn about type widening. I had that at first but decided to keep the original code as-is, but I can change it.

Reply:
As for giving an error, I don’t think it is an error. Especially not in this case where you just can’t quite format the entire space so there’s a small deviation.

However even in the case that you have an 64 MB drive but just want one volume on it without partition table, I think that’s legit. Maybe an informational message would be nice, but I couldn’t find a simple way to do that (as simple as throwing). Or a way to query the used vs available space (a new `info` command of sorts), but that’s a separate feature.